x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/index.ts

```diff
@@ -87,7 +87,7 @@ export class APMEventClient {
   private readonly esClient: ElasticsearchClient;
   private readonly debug: boolean;
   private readonly request: KibanaRequest;
-  private readonly indices: ApmIndicesConfig;
+  public readonly indices: ApmIndicesConfig;
```

Comment:
Making this public wouldn't give the ability to the caller to change the indices?

Reply:
it's readonly no?

Reply:
ah I guess you can't change the identity of the map but you can still override values?

Reply:
Yes, IIRC

Reply:
Good point, so a getter is what you suggest? `apmEventClient.getIndicies()`

Reply:
Either a getter or just use `getAPMIndices()`

Reply:
We can also make ApmIndicesConfig deeply readonly. `getApmIndices()` would also mean recreating a new object every time. Not that it matters much in terms of performance but IMHO a type change should cover this as well:

```ts
export type ApmIndicesConfig = Readonly<{
  sourcemap: string;
  error: string;
  onboarding: string;
  span: string;
  transaction: string;
  metric: string;
  apmAgentConfigurationIndex: string;
  apmCustomLinkIndex: string;
}>;
```

Reply:
> We can also make ApmIndicesConfig deeply readonly. `getApmIndices()` would also mean recreating a new object every time. Not that it matters much in terms of performance but IMHO a type change should cover this as well:
> 
> ```ts
> export type ApmIndicesConfig = Readonly<{
>   sourcemap: string;
>   error: string;
>   onboarding: string;
>   span: string;
>   transaction: string;
>   metric: string;
>   apmAgentConfigurationIndex: string;
>   apmCustomLinkIndex: string;
> }>;
> ```

I'm good with making it readonly. Sounds like a simple change.

Reply:
> Either a getter or just use `getAPMIndices()`

Right now we are passing `indices` around but it feels like it should be available from the client since it's already calling `getAPMIndices` internally.